compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -215,9 +260,65 @@ bool TaggedShapeAnalyzer::verify_tag() const
   return true;
 }
 
+/**
+ * @brief Initialize the class member and checks under conditions
+ *
+ *  Condtiions that have to be met for analyzer
+ *    c1: input rank >= output rank
+ *    c2: The 'perm' of tranpose should be a CircleConst* type
+ *    c3: The shapes of the given nodes should be all known
+ *
+ * @return True, if all conditions are satisfied and class members are initialized successfully
+ *         False, otherwise
+ */
+template <loco::DataType DType>
+bool TaggedShapeAnalyzer::init(const luci::CircleTranspose *front_transpose,
+                               const luci::CircleReshape *mid_reshape,
+                               const luci::CircleTranspose *back_transpose)
```

Comment:
(note) 
This is the main change. I'd like to introduce this `init()` function to explicitly check the analyzer working condition. 

* Before 
   * conditions(`c1` and `c2`)  are checked in callers and some conditions(`c3`) are not checked 
  
* After
   * all conditions are checked in this `init()` function. 
   * if `init()` function fails it means that - the analyzer didn't support this case